packages/frontend/src/store/timeEntries.ts

```diff
@@ -219,3 +239,105 @@ function updateTimeEntries(state: State, paramEntries: FrontendTimentrie[]) {
   }
   state.timeEntries = newTimeEntries;
 }
+
+function getRelevantMonths(): Date[] {
+  const today = new Date();
+  const lastMonth = new Date(today.getFullYear(), today.getMonth() - 1);
+  const twoMonthsAgo = new Date(
+    lastMonth.getFullYear(),
+    lastMonth.getMonth() - 1
+  );
+  const relevantMonths = [twoMonthsAgo, lastMonth, today];
+  return relevantMonths;
+}
+
+function filterTimeEntriesByRelevantMonths(
+  allTimeEntriesAllTasks: TimeEntriesDateFormated[][]
+) {
+  return allTimeEntriesAllTasks.map(timeEntriesSingleTask => {
+    return timeEntriesSingleTask.filter(x => {
+      return getRelevantMonths().some(y => {
+        if (y.getMonth() == x.date.getMonth()) {
+          return y;
+        }
+      });
+    });
+  });
+}
+
+function convertStringToDateAndValueToNumber(
+  timeEntries: FrontendTimentrie[] | null
+) {
+  var newArray: TimeEntriesDateFormated[] = timeEntries
+    ? timeEntries.map((x: FrontendTimentrie) => {
+        let newObj = {
+          id: x.id,
+          value: Number(x.value),
+          taskId: x.taskId,
+          date: new Date(x.date),
+        };
+        return newObj;
+      })
+    : [];
+  return newArray;
+}
```

Comment:
Denne funksjonen syntes jeg også fortjener en test og en omskriving. Bruk gjerne const istendenfo var og let når variablene ikke endres. Bruk gjerne en vanlig if else block istenden for ? :. Det gjør det lettere å lese. Utestet forslag 👇. Vurder om det går an å skrive om koden slik at det aldri vil bli sendt inn null her.
```ts
function convertStringToDateAndValueToNumber(
  timeEntries: FrontendTimentrie[] | null
) {
  const mappedTimeEntries = timeEntries.map(entrie => ({
      id: entrie.id,
      value: Number(entrie.value),
      taskId: entrie.taskId,
      date: new Date(entrie.date),
    }));

  return timeEntries ? mappedTimeEntries : [];
}
```